src/container.ts

```diff
@@ -609,13 +613,17 @@ export class Container {
       throw new Error('Attempted to construct a container from a Container object.');
     }
 
+    const defaultProbeConfiguration: probe.Probe = probe.Probe.fromTcpSocket({
+      port: props.port,
+    });
+
     this.name = props.name ?? 'main';
     this.image = props.image;
     this.port = props.port;
     this._command = props.command;
     this._args = props.args;
-    this._readiness = props.readiness;
-    this._liveness = props.liveness;
+    this._readiness = props.readiness ?? ( props.port ? defaultProbeConfiguration : undefined);
+    this._liveness = props.liveness ?? ( props.port ? defaultProbeConfiguration : undefined);
```

Comment:
@vinayak-kukreja After some more thought I think we shouldn't be configuring default liveness / readiness probes. 

I fear that in many cases, these default probes will not actually represent a healthy container, and by applying them, we are masking a would be validation violation that forces the application author to properly think about these probes. 

I do think though that we can use the TCP probe for the `startupProbe`, since a container is definitely not ready before its port is bounded. The `startupProbe` is safer because it doesn't play a role in the ongoing lifecycle of the container.

Reply:
I see, makes sense. Will update this.